bundles/org.openhab.ui/web/src/pages/settings/things/thing-details.vue

```diff
@@ -371,10 +371,12 @@ export default {
             this.configDirty = false
             this.thingDirty = false
 
-            // special treatment for Z-Wave actions
-            if (this.thingType.UID.indexOf('zwave') === 0) {
-              this.zwaveActions = this.configDescriptions.parameters.filter((p) => p.groupName === 'actions')
-              this.configDescriptions.parameters = this.configDescriptions.parameters.filter((p) => p.groupName !== 'actions')
+            // special treatment for actions (heuristic match by group name/label)
+            if (this.configDescriptions.parameterGroups.filter((pg) => pg.name === 'actions').every((pg) => pg.label === 'Actions' && pg.description === 'Actions')) {
```

Comment:
This heuristic is good enough but since parameter groups have a "context" attribute:

https://next.openhab.org/javadoc/latest/org/openhab/core/config/core/dto/configdescriptionparametergroupdto

I'm wondering whether this could perhaps be the way to go - look for e.g. an "actions" context in the parameter group definition instead of checking for labels & descriptions - it certainly would be more elegant IMO.

We could still make an exception for the Z-Wave things unless @cdjackson agrees to add the "actions" context for the actions parameter group for them.

I'm not sure how easy that is to add such contexts to parameter groups, whether in declarative or imperative form.

Reply:
Indeed, there seems to be a "param group context" defined, though not documented too widely. Per [this](context) it would seem it may be a freeform string (the per-param context seems to be a pre-defined list).
Let me try to see how easy it is to set/use and report back (may take me about a week to do so, though)...

I 100% agree it would be more elegant than this heuristic.
That said, I think the ROI is there when ZWave binding also adopts it, otherwise we're back into heuristic-land :). Not sure what'd be the timeline, but how about a 2-step approach? I.e. this heuristic *for now* (possibly with an "or" for a by-context match) and follow through with a proper update if/once ZWave binding changes?

Reply:
> I.e. this heuristic for now (possibly with an "or" for a by-context match) and follow through with a proper update if/once ZWave binding changes?

Works for me!

Reply:
The "by-context" method does work indeed, so I added it as primary.
Any param group constructed with `.withContext("actions")` will now be picked as action container.

As per our discussion, I've left the heuristic match by `name` and `label` for compat with today's Z-wave binding. 
Note I had to pull the "by description" criterion, since Z-Wave doesn't use it uniformly and only sets it on non-controller devices...

--
While at it, I attempted to make it a bit more generic (hopefully the code explains the intent well enough). Do ask if it raises any concerns.